File: ios/FreelanceKar/Views/Sheets/HireSheet.swift
import SwiftUI

struct HireSheetView: View {
    @EnvironmentObject var store: AppStore

    var body: some View {
        let s = store.state
        let pro = Selectors.activePro(s)
        let pkg = Selectors.selectedPackage(s)

        FKSheet(isOpen: s.hire > 0, onClose: { store.closeHire() }, maxHeightFraction: 0.88) {
            switch s.hire {
            case 1: detailsStep(s, pro, pkg)
            case 2: payStep(s, pro, pkg)
            case 3: successStep(s, pro, pkg)
            case 4: failureStep(pro, pkg)
            default: EmptyView()
            }
        }
    }

    @ViewBuilder private func detailsStep(_ s: AppState, _ pro: Pro, _ pkg: Package) -> some View {
        VStack(alignment: .leading, spacing: 0) {
            HStack {
                Text("Hire \(pro.first)").font(.system(size: 18, weight: .semibold))
                Spacer()
                Button { store.closeHire() } label: { Image(systemName: "xmark").font(.system(size: 14)).foregroundStyle(FK.text50) }
            }

            Group2(label: "SERVICE") {
                ForEach(Array(pro.packages.enumerated()), id: \.element.id) { i, p in
                    Button { store.pickService(i) } label: {
                        HStack {
                            VStack(alignment: .leading, spacing: 2) {
                                Text(p.name).font(.system(size: 13, weight: .semibold))
                                Text(p.terms).font(.system(size: 10.5)).foregroundStyle(FK.text50)
                            }
                            Spacer()
                            Text(p.price).font(.system(size: 12.5, weight: .semibold))
                        }
                        .padding(.horizontal, 13).padding(.vertical, 12)
                    }
                    .foregroundStyle(FK.text)
                    .background(FK.nested)
                    .clipShape(RoundedRectangle(cornerRadius: 13, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: 13).stroke(i == s.svcIdx ? FK.lime : FK.line08, lineWidth: 1))
                }
            }

            Group2(label: "START DATE") {
                HStack(spacing: 7) {
                    ForEach(Array(Seed.dateLabels.enumerated()), id: \.offset) { i, d in
                        Chip(title: d, on: s.hireDateIdx == i) { store.setHireDate(i) }
                    }
                }
            }

            Group2(label: "BRIEF") {
                FKTextArea(placeholder: "What exactly do you need? Add links or references.",
                           text: Binding(get: { s.hireNote }, set: { store.setHireNote($0) }), minHeight: 64)
                if s.hireError {
                    Text("Add a line about the work so \(pro.first) can confirm.").font(.system(size: 11.5, weight: .medium)).foregroundStyle(FK.alert)
                }
            }

            PrimaryButton(title: "Continue · \(pkg.price)") { store.toPay() }.padding(.top, 18)
        }
        .padding(.horizontal, 20).padding(.top, 18).padding(.bottom, 22)
    }

    @ViewBuilder private func payStep(_ s: AppState, _ pro: Pro, _ pkg: Package) -> some View {
        VStack(alignment: .leading, spacing: 0) {
            HStack {
                Button { store.toDetails() } label: { Image(systemName: "chevron.left").font(.system(size: 15)).foregroundStyle(FK.text60) }
                Spacer()
                Text("Secure payment").font(.fkScreenTitle)
                Spacer()
                Button { store.closeHire() } label: { Image(systemName: "xmark").font(.system(size: 14)).foregroundStyle(FK.text50) }
            }

            VStack(spacing: 0) {
                HStack(spacing: 11) {
                    Stripe(cornerRadius: 11).frame(width: 38, height: 38)
                    VStack(alignment: .leading, spacing: 2) {
                        Text(pro.name).font(.system(size: 13, weight: .semibold))
                        Text(pkg.name).font(.system(size: 11)).foregroundStyle(FK.text50)
                    }
                    Spacer()
                }
                .padding(.bottom, 12)
                .overlay(Rectangle().frame(height: 1).foregroundStyle(FK.line07), alignment: .bottom)

                HStack {
                    Text("Service").foregroundStyle(FK.text60)
                    Spacer()
                    Text(pkg.price).foregroundStyle(FK.text)
                }.font(.system(size: 12.5)).padding(.top, 12)

                HStack {
                    Text("Total to pay \(pro.first)")
                    Spacer()
                    Text(money(pkg.amount))
                }
                .font(.system(size: 14, weight: .semibold))
                .padding(.top, 11)
                .overlay(Rectangle().frame(height: 1).foregroundStyle(FK.line07), alignment: .top)
                .padding(.top, 11)
            }
            .padding(14)
            .background(FK.nested)
            .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line09, lineWidth: 1))
            .padding(.top, 16)

            Group2(label: "PAY WITH") {
                ForEach(Array(Seed.payMethods.enumerated()), id: \.offset) { i, m in
                    Button { store.setMethod(i) } label: {
                        HStack(spacing: 11) {
                            Text(m.0).font(.system(size: 14))
                            VStack(alignment: .leading, spacing: 2) {
                                Text(m.1).font(.system(size: 12.5, weight: .semibold))
                                Text(m.2).font(.system(size: 10.5)).foregroundStyle(FK.text50)
                            }
                            Spacer()
                            Text(i == s.methodIdx ? "●" : "○").font(.system(size: 12)).foregroundStyle(i == s.methodIdx ? FK.lime : FK.text35)
                        }
                        .padding(.horizontal, 13).padding(.vertical, 12)
                    }
                    .foregroundStyle(FK.text)
                    .background(FK.nested)
                    .clipShape(RoundedRectangle(cornerRadius: 13, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: 13).stroke(i == s.methodIdx ? FK.lime : FK.line09, lineWidth: 1))
                }
            }

            Text("You pay \(pro.first) directly — no platform fee. This confirms the booking and shares your contact details.")
                .font(.system(size: 11.5)).foregroundStyle(FK.text45).lineSpacing(3).padding(.top, 14)

            PrimaryButton(title: s.paying ? "Confirming…" : "Confirm booking", disabled: s.paying) {
                store.payNow(pro, pkg.name, pkg.amount)
            }.padding(.top, 16)
        }
        .padding(.horizontal, 20).padding(.top, 18).padding(.bottom, 22)
    }

    @ViewBuilder private func successStep(_ s: AppState, _ pro: Pro, _ pkg: Package) -> some View {
        VStack(spacing: 0) {
            Circle().fill(FK.lime).frame(width: 52, height: 52)
                .overlay(Text("✓").font(.system(size: 22)).foregroundStyle(FK.onLime))
            Text("Booked with \(pro.first)").font(.system(size: 19, weight: .semibold)).padding(.top, 16)
            Text("Pay \(pro.first) \(money(pkg.amount)) directly on delivery. Usually confirms within \(pro.replyIn) — you’ll get a notification.")
                .font(.system(size: 13)).foregroundStyle(FK.text55).multilineTextAlignment(.center).lineSpacing(4).padding(.top, 8)

            VStack(alignment: .leading, spacing: 9) {
                DetailRow(label: "Service", value: pkg.name)
                DetailRow(label: "Starts", value: Seed.dateLabels[s.hireDateIdx])
                DetailRow(label: "Order", value: s.bookings.first?.id ?? "")
            }
            .padding(14)
            .background(FK.nested)
            .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line09, lineWidth: 1))
            .padding(.top, 18)

            HStack(spacing: 9) {
                Button { store.afterBookChat() } label: {
                    Text("Open chat").font(.system(size: 13, weight: .medium)).foregroundStyle(FK.text)
                        .frame(maxWidth: .infinity).padding(13)
                }
                .overlay(RoundedRectangle(cornerRadius: 12).stroke(FK.line16, lineWidth: 1))
                Button { store.afterBookHome() } label: {
                    Text("Done").font(.system(size: 13, weight: .semibold)).foregroundStyle(FK.onLime)
                        .frame(maxWidth: .infinity).padding(13)
                }
                .background(FK.lime)
                .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
            }.padding(.top, 16)
        }
        .padding(.horizontal, 22).padding(.top, 34).padding(.bottom, 26)
    }

    @ViewBuilder private func failureStep(_ pro: Pro, _ pkg: Package) -> some View {
        VStack(spacing: 0) {
            Circle().fill(FK.alert14).frame(width: 52, height: 52)
                .overlay(Circle().stroke(FK.alertBd50, lineWidth: 1))
                .overlay(Text("!").font(.system(size: 22)).foregroundStyle(FK.alert))
            Text("Payment didn’t go through").font(.system(size: 19, weight: .semibold)).padding(.top, 16)
            Text("Your bank declined the card. Nothing was charged and \(pro.first) hasn’t been booked yet.")
                .font(.system(size: 13)).foregroundStyle(FK.text55).multilineTextAlignment(.center).lineSpacing(4).padding(.top, 8)

            VStack(alignment: .leading, spacing: 7) {
                DetailRow(label: "Attempted", value: money(pkg.amount))
                DetailRow(label: "Reason", value: "DECLINED_BY_ISSUER")
            }
            .padding(.horizontal, 14).padding(.vertical, 13)
            .background(FK.nested)
            .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line09, lineWidth: 1))
            .padding(.top, 18)

            VStack(spacing: 9) {
                Button { store.payWithUpi() } label: {
                    Text("Pay with UPI instead").font(.system(size: 13.5, weight: .semibold)).foregroundStyle(FK.onLime)
                        .frame(maxWidth: .infinity).padding(14)
                }
                .background(FK.lime)
                .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
                Button { store.retryPay() } label: {
                    Text("Try the card again").font(.system(size: 13, weight: .medium)).foregroundStyle(FK.text)
                        .frame(maxWidth: .infinity).padding(13)
                }
                .overlay(RoundedRectangle(cornerRadius: 12).stroke(FK.line16, lineWidth: 1))
            }.padding(.top, 16)
        }
        .padding(.horizontal, 22).padding(.top, 32).padding(.bottom, 26)
    }
}

private struct Group2<Content: View>: View {
    var label: String
    @ViewBuilder var content: Content
    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            MonoLabel(text: label)
            content
        }.padding(.top, 16)
    }
}

private struct DetailRow: View {
    var label: String
    var value: String
    var body: some View {
        HStack {
            Text(label).foregroundStyle(FK.text55)
            Spacer()
            Text(value).fontWeight(.medium).foregroundStyle(FK.text)
        }.font(.system(size: 12))
    }
}

File: android/app/src/main/java/com/freelancekar/app/ui/sheets/HireSheet.kt
package com.freelancekar.app.ui.sheets

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.material3.TextFieldDefaults
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.*
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.Chip
import com.freelancekar.app.ui.components.PrimaryButton
import com.freelancekar.app.ui.components.Stripe
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun HireSheet(state: AppState, vm: AppViewModel) {
    if (state.hire <= 0) return
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
    val pro = activePro(state)
    val pkg = selectedPackage(state)

    ModalBottomSheet(
        onDismissRequest = { vm.closeHire() },
        sheetState = sheetState,
        containerColor = FkColor.sheet,
    ) {
        when (state.hire) {
            1 -> HireDetails(state, vm, pro, pkg)
            2 -> HirePay(state, vm, pro, pkg)
            3 -> HireSuccess(state, vm, pro, pkg)
            4 -> HireFailure(state, vm, pro, pkg)
        }
    }
}

@Composable
private fun Group(label: String, content: @Composable androidx.compose.foundation.layout.ColumnScope.() -> Unit) {
    Column(Modifier.padding(top = 16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(label, style = FkType.monoLabel.copy(color = FkColor.text45))
        content()
    }
}

@Composable
private fun DetailRow(label: String, value: String) {
    Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
        Text(label, style = FkType.meta.copy(color = FkColor.text55))
        Text(value, style = FkType.meta.copy(color = FkColor.text, fontWeight = androidx.compose.ui.text.font.FontWeight.Medium))
    }
}

@Composable
private fun HireDetails(state: AppState, vm: AppViewModel, pro: Pro, pkg: Package) {
    Column(Modifier.fillMaxWidth().padding(horizontal = 20.dp).padding(bottom = 22.dp)) {
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Text("Hire ${pro.first}", style = FkType.h1sm.copy(fontSize = androidx.compose.ui.unit.TextUnit(18f, androidx.compose.ui.unit.TextUnitType.Sp)))
            Text("✕", style = FkType.body.copy(color = FkColor.text50), modifier = Modifier.clickable { vm.closeHire() })
        }

        Group("SERVICE") {
            pro.packages.forEachIndexed { i, p ->
                Row(
                    Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(13.dp))
                        .background(FkColor.nested)
                        .border(1.dp, if (i == state.svcIdx) FkColor.lime else FkColor.line08, RoundedCornerShape(13.dp))
                        .clickable { vm.pickService(i) }
                        .padding(horizontal = 13.dp, vertical = 12.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                ) {
                    Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
                        Text(p.name, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                        Text(p.terms, style = FkType.metaXs.copy(color = FkColor.text50))
                    }
                    Text(p.price, style = FkType.bodySm.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                }
            }
        }

        Group("START DATE") {
            Row(horizontalArrangement = Arrangement.spacedBy(7.dp)) {
                DATE_LABELS.forEachIndexed { i, label ->
                    Box(Modifier.weight(1f)) {
                        Chip(on = state.hireDateIdx == i, onClick = { vm.setHireDate(i) }, text = label, modifier = Modifier.fillMaxWidth())
                    }
                }
            }
        }

        Group("BRIEF") {
            TextField(
                value = state.hireNote,
                onValueChange = { vm.setHireNote(it) },
                placeholder = { Text("What exactly do you need? Add links or references.", style = FkType.body.copy(color = FkColor.text42)) },
                modifier = Modifier
                    .fillMaxWidth()
                    .heightIn(min = 64.dp)
                    .clip(RoundedCornerShape(13.dp))
                    .border(1.dp, FkColor.line09, RoundedCornerShape(13.dp)),
                colors = TextFieldDefaults.colors(
                    focusedContainerColor = FkColor.nested, unfocusedContainerColor = FkColor.nested,
                    focusedIndicatorColor = Color.Transparent, unfocusedIndicatorColor = Color.Transparent,
                ),
                textStyle = FkType.body.copy(color = FkColor.text),
            )
            if (state.hireError) {
                Text("Add a line about the work so ${pro.first} can confirm.", style = FkType.meta.copy(color = FkColor.alert))
            }
        }

        PrimaryButton({ vm.toPay() }, modifier = Modifier.padding(top = 18.dp), text = "Continue · ${pkg.price}")
    }
}

@Composable
private fun HirePay(state: AppState, vm: AppViewModel, pro: Pro, pkg: Package) {
    Column(Modifier.fillMaxWidth().padding(horizontal = 20.dp).padding(bottom = 22.dp)) {
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
            Text("←", style = FkType.body.copy(color = FkColor.text60), modifier = Modifier.clickable { vm.toDetails() })
            Text("Secure payment", style = FkType.screenTitle)
            Text("✕", style = FkType.body.copy(color = FkColor.text50), modifier = Modifier.clickable { vm.closeHire() })
        }

        Column(
            Modifier
                .fillMaxWidth()
                .padding(top = 16.dp)
                .clip(RoundedCornerShape(15.dp))
                .background(FkColor.nested)
                .border(1.dp, FkColor.line09, RoundedCornerShape(15.dp))
                .padding(14.dp),
        ) {
            Row(
                Modifier.fillMaxWidth().padding(bottom = 12.dp).border(0.dp, FkColor.line07),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Stripe(Modifier.size(38.dp).clip(RoundedCornerShape(11.dp)))
                Spacer(Modifier.width(11.dp))
                Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
                    Text(pro.name, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                    Text(pkg.name, style = FkType.meta.copy(color = FkColor.text50))
                }
            }
            Row(Modifier.fillMaxWidth().padding(top = 12.dp), horizontalArrangement = Arrangement.SpaceBetween) {
                Text("Service", style = FkType.bodySm.copy(color = FkColor.text60))
                Text(pkg.price, style = FkType.bodySm.copy(color = FkColor.text))
            }
            Row(
                Modifier.fillMaxWidth().padding(top = 11.dp).border(0.dp, FkColor.line07).padding(top = 11.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
            ) {
                Text("Total to pay ${pro.first}", style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                Text(money(pkg.amount), style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
            }
        }

        Group("PAY WITH") {
            PAY_METHODS.forEachIndexed { i, (icon, name, note) ->
                Row(
                    Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(13.dp))
                        .background(FkColor.nested)
                        .border(1.dp, if (i == state.methodIdx) FkColor.lime else FkColor.line09, RoundedCornerShape(13.dp))
                        .clickable { vm.setMethod(i) }
                        .padding(horizontal = 13.dp, vertical = 12.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text(icon, style = FkType.body)
                    Spacer(Modifier.width(11.dp))
                    Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                        Text(name, style = FkType.bodySm.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                        Text(note, style = FkType.metaXs.copy(color = FkColor.text50))
                    }
                    Text(if (i == state.methodIdx) "●" else "○", style = FkType.meta.copy(color = if (i == state.methodIdx) FkColor.lime else FkColor.text35))
                }
            }
        }

        Text(
            "You pay ${pro.first} directly — no platform fee. This confirms the booking and shares your contact details.",
            style = FkType.meta.copy(color = FkColor.text45),
            modifier = Modifier.padding(top = 14.dp),
        )

        PrimaryButton(
            { vm.payNow(pro, pkg.name, pkg.amount) },
            modifier = Modifier.padding(top = 16.dp),
            enabled = !state.paying,
            text = if (state.paying) "Confirming…" else "Confirm booking",
        )
    }
}

@Composable
private fun HireSuccess(state: AppState, vm: AppViewModel, pro: Pro, pkg: Package) {
    Column(
        Modifier.fillMaxWidth().padding(horizontal = 22.dp).padding(top = 34.dp, bottom = 26.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Box(
            Modifier.size(52.dp).clip(CircleShape).background(FkColor.lime),
            contentAlignment = Alignment.Center,
        ) { Text("✓", style = FkType.h2.copy(color = FkColor.onLime)) }
        Spacer(Modifier.height(16.dp))
        Text("Booked with ${pro.first}", style = FkType.h1sm)
        Spacer(Modifier.height(8.dp))
        Text(
            "Pay ${pro.first} ${money(pkg.amount)} directly on delivery. Usually confirms within ${pro.replyIn} — you'll get a notification.",
            style = FkType.body.copy(color = FkColor.text55),
            textAlign = androidx.compose.ui.text.style.TextAlign.Center,
        )
        Column(
            Modifier
                .fillMaxWidth()
                .padding(top = 18.dp)
                .clip(RoundedCornerShape(15.dp))
                .background(FkColor.nested)
                .border(1.dp, FkColor.line09, RoundedCornerShape(15.dp))
                .padding(14.dp),
            verticalArrangement = Arrangement.spacedBy(9.dp),
        ) {
            DetailRow("Service", pkg.name)
            DetailRow("Starts", DATE_LABELS[state.hireDateIdx])
            DetailRow("Order", state.bookings.firstOrNull()?.id ?: "")
        }
        Row(Modifier.fillMaxWidth().padding(top = 16.dp), horizontalArrangement = Arrangement.spacedBy(9.dp)) {
            Box(
                Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(12.dp))
                    .border(1.dp, FkColor.line16, RoundedCornerShape(12.dp))
                    .clickable { vm.afterBookChat() }
                    .padding(13.dp),
            ) { Text("Open chat", style = FkType.bodySm.copy(color = FkColor.text), modifier = Modifier.fillMaxWidth(), textAlign = androidx.compose.ui.text.style.TextAlign.Center) }
            Box(
                Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(12.dp))
                    .background(FkColor.lime)
                    .clickable { vm.afterBookHome() }
                    .padding(13.dp),
            ) { Text("Done", style = FkType.bodySm.copy(color = FkColor.onLime, fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold), modifier = Modifier.fillMaxWidth(), textAlign = androidx.compose.ui.text.style.TextAlign.Center) }
        }
    }
}

@Composable
private fun HireFailure(state: AppState, vm: AppViewModel, pro: Pro, pkg: Package) {
    Column(
        Modifier.fillMaxWidth().padding(horizontal = 22.dp).padding(top = 32.dp, bottom = 26.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Box(
            Modifier
                .size(52.dp)
                .clip(CircleShape)
                .background(FkColor.alert14)
                .border(1.dp, FkColor.alertBd50, CircleShape),
            contentAlignment = Alignment.Center,
        ) { Text("!", style = FkType.h2.copy(color = FkColor.alert)) }
        Spacer(Modifier.height(16.dp))
        Text("Payment didn't go through", style = FkType.h1sm)
        Spacer(Modifier.height(8.dp))
        Text(
            "Your bank declined the card. Nothing was charged and ${pro.first} hasn't been booked yet.",
            style = FkType.body.copy(color = FkColor.text55),
            textAlign = androidx.compose.ui.text.style.TextAlign.Center,
        )
        Column(
            Modifier
                .fillMaxWidth()
                .padding(top = 18.dp)
                .clip(RoundedCornerShape(15.dp))
                .background(FkColor.nested)
                .border(1.dp, FkColor.line09, RoundedCornerShape(15.dp))
                .padding(horizontal = 14.dp, vertical = 13.dp),
            verticalArrangement = Arrangement.spacedBy(7.dp),
        ) {
            DetailRow("Attempted", money(pkg.amount))
            DetailRow("Reason", "DECLINED_BY_ISSUER")
        }
        Column(Modifier.fillMaxWidth().padding(top = 16.dp), verticalArrangement = Arrangement.spacedBy(9.dp)) {
            Box(
                Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(12.dp))
                    .background(FkColor.lime)
                    .clickable { vm.payWithUpi() }
                    .padding(14.dp),
            ) { Text("Pay with UPI instead", style = FkType.body.copy(color = FkColor.onLime, fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold), modifier = Modifier.fillMaxWidth(), textAlign = androidx.compose.ui.text.style.TextAlign.Center) }
            Box(
                Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(12.dp))
                    .border(1.dp, FkColor.line16, RoundedCornerShape(12.dp))
                    .clickable { vm.retryPay() }
                    .padding(13.dp),
            ) { Text("Try the card again", style = FkType.body.copy(color = FkColor.text), modifier = Modifier.fillMaxWidth(), textAlign = androidx.compose.ui.text.style.TextAlign.Center) }
        }
    }
}
